flytekit/core/tracker.py

```diff
@@ -92,6 +137,29 @@ def find_lhs(self) -> str:
                 # continue looping through m.
                 logger.warning("Caught ValueError {} while attempting to auto-assign name".format(err))
 
+        # try to find object in module when the tracked instance is defined in the __main__ module
+        module = import_module_from_file(self._instantiated_in, self._module_file)
+
+        def _candidate_name_matches(candidate) -> bool:
+            if not hasattr(candidate, "name") or not hasattr(self, "name"):
+                return False
+            return candidate.name == self.name
+
+        for k in dir(module):
```

Comment:
are we looking for an exact object match here?  why can't we just do object id equality if that's the case?

Reply:
because `module` here is now a different module from `main`, task/dynamic/workflow functions can no longer be identified via equality checks.